src/hotspot/share/runtime/deoptimization.hpp

```diff
@@ -159,7 +159,7 @@ class Deoptimization : AllStatic {
 JVMCI_ONLY(public:)
 
   // Support for restoring non-escaping objects
-  static bool realloc_objects(JavaThread* thread, frame* fr, GrowableArray<ScopeValue*>* objects, TRAPS);
```

Comment:
It would be better keep this method and add 2nd method instead of direct replacement.